### PR TITLE
Move shouldDiscover in a new system/Modules class

### PR DESCRIPTION
**Pull request by mostafakhudair**
Creating new system/Modules/Modules.php file and moves shouldDiscover function into it away from user hand like before app/Config/Modules.php
to help avoid tampering with the core functions

### CODE REVIEW

File: system/bootstrap.php [hunk 1]
```diff
@@ -120,6 +120,7 @@
 {
 	require_once SYSTEMPATH . 'Config/AutoloadConfig.php';
 	require_once APPPATH . 'Config/Autoload.php';
+	require_once SYSTEMPATH . 'Modules/Modules.php';
```
Comment (samsonasik) [inline system/bootstrap.php:123]: should be added to `system/Test/bootstrap.php` as well.

**Comment by lonnieezell** [comment]
@mostafakhudair Are you going to be able to fix up these tests and address the comment that @samsonasik made?

**Comment by samsonasik** [comment]
@lonnieezell @mostafakhudair already applied to `system/Test/bootstrap.php`

**Comment by samsonasik** [comment]
The tests still needs update indeed

**Comment by mostafakhudair** [comment]
@lonnieezell everthing is wroking well now

**Pull request merged**